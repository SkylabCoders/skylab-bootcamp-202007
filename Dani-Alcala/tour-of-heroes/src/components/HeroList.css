.list-items {
    display: flex;
    flex-direction: row;
}