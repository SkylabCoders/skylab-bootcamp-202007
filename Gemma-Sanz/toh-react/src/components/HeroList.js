import React, { useState, useEffect } from 'react';
import { Link } from "react-router-dom";
import '../styles.css';
import heroStore from '../stores/heroStore';
import { loadHeroes, deleteHero } from '../actions/heroActions';



function HeroList(props) {
    const [heroes, setHeroes] = useState([])

    useEffect(() => {
        heroStore.addChangeListener(onChange);
        if (heroes.length === 0) loadHeroes();
        return () => heroStore.removeChangeListener(onChange);
    }, [heroes.length]);

    function onChange() {
        setHeroes(heroStore.getHeroes());
    }

    function onDelete(event, heroId) {
        event.preventDefault();
        deleteHero(heroId);
    }

    return (
        <div className="mainContainer__list containerComponent">
            <>
                {heroes.map((hero) => (
                    <li key={hero.id}>
                        < Link key={hero.id} to={`/hero/${hero.id}`}>
                            <div className="button__list" >
                                <span className="element__list__id">{hero.id}</span>
                                <span className="element__list__name">{hero.name}</span>
                            </div>
                        </Link>
                        <div className="hero-list__item--delete">
                            <button onClick={(event) => onDelete(event, hero.id)}>X</button>
                        </div>
                    </li>
                ))}
            </>
        </div >
    );
}


export default HeroList;