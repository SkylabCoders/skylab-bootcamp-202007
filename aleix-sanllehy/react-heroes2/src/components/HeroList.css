.hero-list__item {
	height: 24px;
}

.hero-list__item:hover .hero-list__item--delete {
	display: inline-block;
}

.hero-list__item--delete {
	display: none;
	margin-left: 12px;
}
