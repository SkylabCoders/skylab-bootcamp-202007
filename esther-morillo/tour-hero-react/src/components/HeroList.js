import React from 'react';
import '../hero-service'
import heroList from '../hero-service';
import HeroListItem from './HeroLIstItem'


function HeroList() {
	return (
		heroList.map((hero) => (
		<HeroListItem
			key={hero.id}
			id={hero.id}
			name={hero.name}
		/>
		))
	)
	
}

export default HeroList;
