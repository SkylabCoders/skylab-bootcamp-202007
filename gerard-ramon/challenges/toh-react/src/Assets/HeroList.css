body {
	margin: auto;
	font-family: Arial, Helvetica, sans-serif;
	margin-left: 50px;
}

.button {
	background-color: #eee;
	border: none;
	padding: 5px 10px;
	border-radius: 4px;
	cursor: pointer;
	cursor: hand;
	margin-right: 5px;
}
.button:hover {
	background-color: #cfd8dc;
}

.heroDashboard {
	display: flex;
	flex-direction: row;
	align-items: center;
}

.heroDashboardLink {
	background-color: #5f7d8c;
	color: whitesmoke;
	font-weight: bold;
	list-style: none;
	margin: 12px;
	padding: 40px;
}
.heroDashboardLink:hover {
	background-color: #cfd8dc;
	cursor: pointer;
}

.containerList {
	display: flex;
	flex-direction: column;
}

.heroContainer {
	display: flex;
	align-items: center;
}

.heroListLink {
	color: rgb(103, 103, 139);
	background-color: #eeeeee;
	margin: 4px;
	padding: 10px 0px;
	border-radius: 5px;
	font-size: 14px;
	width: 15%;
	min-width: 120px;
	font-weight: bold;
	text-decoration: none;
}

.heroId {
	background-color: #5f7d8c;
	color: whitesmoke;
	padding: 10px;
	border-top-left-radius: 5px;
	border-bottom-left-radius: 5px;
}

.listButton {
	margin: 5px;
}
