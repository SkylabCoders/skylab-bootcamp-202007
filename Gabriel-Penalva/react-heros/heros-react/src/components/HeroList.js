import React, { useState, useEffect } from 'react';
import './HeroList.css'
import heroStore from '../stores/heroStore';
import { loadHeros } from '../actions/heroActions';

function HeroList() {
    const [heros, setHeros] = useState([]);

    useEffect(() => {
        heroStore.addChangeListener(onChange);
        if (heros.length === 0) {
            loadHeros();
        }
        return () => heroStore.removeChangeListerner(onChange);
    }, [heros.length]);

    function onChange() {
        setHeros(heroStore.getHeroes())
    }
    return (
        <div>
            <ul className='heroes'>
                <h3>List of Heros</h3>
                {heros.map((hero) => (
                    <li key={hero.id}><a href={`/id/${hero.id}`}
                        className="hero-list_item"><span className="badge ">{hero.id}</span> {hero.name}</a></li>
                ))}
            </ul>
        </div>)
}




export default HeroList;