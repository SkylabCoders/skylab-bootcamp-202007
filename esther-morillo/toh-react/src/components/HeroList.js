import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import heroStore from '../stores/heroStore';
import { loadHeroes, deleteHero } from '../actions/heroActions';
import './HeroList.css';

function HeroList() {
	const [heroes, setHeroes] = useState(heroStore.getHeroes());

	useEffect(() => {
		//escucha, cuando haya un cambio en el store ejecuta, pero no la ejecuto de primeras (y me lleva a la función)
		heroStore.addChangeListener(onChange);
		//cada vez que entre aquí hace una petición para actulizar la lista de hérores. Aquí es cuando entro y solo si la lista está vacía (que será porque no hemos hecho la petición)
		if (heroes.length === 0) loadHeroes();
		return () => heroStore.removeChangeListener(onChange);
	}, [heroes.length]);

	function onChange() {
		setHeroes(heroStore.getHeroes()); //está en el store, retorna la lista 
	}

	function onDelete(event, heroId) {
		event.preventDefault();
		//Lo único que hace es borrar desde otra función que está en el ACTIONS
		deleteHero(heroId);
	}

	return (
		<ul>
			{heroes.map((hero) => (
				<li key={hero.id} className="hero-list__item">
					<Link to={`/hero/${hero.id}`}>
						{hero.id}: {hero.name}
					</Link>

					{/* Agregamos botón y llamamos a una función */}
					<div className="hero-list__item--delete">
						<button onClick={(event) => onDelete(event, hero.id)}>X</button>
					</div>
				</li>
			))}
		</ul>
	);
}

export default HeroList;
