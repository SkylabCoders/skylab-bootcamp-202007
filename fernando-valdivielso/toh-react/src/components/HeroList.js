import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import heroStore from '../stores/heroStore';
import { loadHeroes, deleteHero } from '../actions/heroActions';
import './HeroList.css';




function HeroList() {
	const [heroes, setHeroes] = useState([]);

	useEffect(() => {
		heroStore.addChangeListener(onChange);
		if (heroes.length === 0) {
			loadHeroes()
		};
		return () => heroStore.removeChangeListener(onChange);
	}, [heroes.length]);

	function onChange() {
		setHeroes(heroStore.getHeroes());
	}

	function onDelete(event, heroId) {
		event.preventDefault();
		deleteHero(heroId);
	}

	return (
		<ul>
			{heroes.map((hero) => (
				<li key={hero.id} className="hero-list__item">
					<Link to={`/hero/${hero.id}`}>
						{hero.id}: {hero.name}
					</Link>
					<div className="hero-list__item--delete">
						<button onClick={(event) => onDelete(event, hero.id)}>X</button>
					</div>
				</li>
			))}
		</ul>
	);
}

export default HeroList;





// import React, { useState, useEffect } from 'react';
// import heroList from '../heroData'
// import { Link } from 'react-router-dom'
// import heroStore from '../stores/heroStore';

// function HeroList() {
// 	const [heroes, setHeroes] = useState(heroList);  //se declara una array para hacer destructuring y a useState se le pasa el valor inicial

// 	useEffect(() => {
// 		heroStore.addChangeListener(onChange)
// 		if (heroes.length === 0);
// 		return () => heroStore.removeChangeListener(onChange);
// 	}, [heroes.length]);  //cada vez que la longitud de la lista cambie, run onChange

// 	function onChange() {
// 		setHeroes(heroStore.getHeroes())
// 	}

// 	return (
// 		<div>
// 			<ul>
// 				{heroes.map(hero => (
// 					<li key={hero.id}>
// 						<Link to={`/hero/${hero.id}`}>{hero.id}: {hero.name} </Link>
// 					</li>
// 				))}
// 			</ul>
// 		</div>
// 	);

// }



// export default HeroList;
