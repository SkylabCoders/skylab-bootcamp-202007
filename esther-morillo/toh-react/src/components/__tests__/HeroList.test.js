import renderer from 'react-test-renderer';
import React from 'react'
import HeroList from '../HeroList';


describe('Detail snapshot', () => {
    const tree = renderer.create(<HeroList />);

    it('should match', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    });
})