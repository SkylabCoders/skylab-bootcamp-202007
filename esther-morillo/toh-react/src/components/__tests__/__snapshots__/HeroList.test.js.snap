// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail snapshot should match 1`] = `<ul />`;
